### ISSUE TITLE
Update Azure Blob Storage to follow MS security best practices

### ISSUE THREAD

**Issue by ShaneCourtrille** [OrchardCMS/OrchardCore#16336]
### Is your feature request related to a problem? Please describe.
We are currently implementing Microsoft best practices for managing access to Azure Blob Storage and we've ran into a problem with the existing OC BlobFileStore.cs due to the fact we cannot easily override how the BlobContainerClient is created.

### Describe the solution you'd like
OC BlobFileStore should follow MS [best practices](https://learn.microsoft.com/en-us/azure/storage/blobs/security-recommendations) which require the usage of either a Microsoft Entra ID or User delegation SAS as a fallback for those who aren't using Entra.

### Describe alternatives you've considered
Our current solution involves adding a new interface which allows us to control how the client is being created like this.  This works well enough for those who are heavily customizing OC but doesn't seem like an appropriate solution for the general usage.

```
public BlobFileStore(IBlobStorageOptions options, IClock clock, IContentTypeProvider contentTypeProvider, IBlobContainerClientFactory blobContainerClientFactory)
        {
            _options = options;
            _clock = clock;
            _contentTypeProvider = contentTypeProvider;
            _blobContainer = blobContainerClientFactory.Create(_options);

            if (!String.IsNullOrEmpty(_options.BasePath))
            {
                _basePrefix = NormalizePrefix(_options.BasePath);
            }
        }
```

**Comment by Piedone** [comment]
Thanks! What exactly would be a solution for you? I.e. what needs to be coded? New configuration options for `BlobFileStore`? BTW you can override `IMediaFileStore` and use a custom one (if this is not what you already do in the code snippet).

**Comment by ShaneCourtrille** [comment]
@Piedone We are going to customize the code so that we can provide our own IBlobContainerClientFactory but it would be better if OC followed MS best practices by supporting Microsoft Entra ID or User delegation SAS token usage out of the box.

**Comment by Piedone** [comment]
OK, thanks, so we're talking about adding new configuration options.

**Comment by sebastienros** [comment]
@MikeAlhayek really wants this to be consistent across features. Maybe we could have a common Authentication section (Azure SDK has a similar thing) that is bound for each feature, and configure the clients with these.

**Comment by rjpowers10** [comment]
I've been trying out [Aspire](https://aspire.dev/) with my Orchard Core app . It had been going smoothly but the way OC instantiates `BlobClient` and `BlobContainerClient` has become a roadblock. OC is based on having a container name and a connection string, but the feeling that I get is Aspire really wants you to use a URI and managed identity. [You can see here that Aspire](https://aspire.dev/integrations/cloud/azure/azure-storage-blobs/#blob-storage) only generates a connection string for emulators, when you publish it generates the URI.

In short, I think if OC allowed for something like this
```cs
_blobContainerClient = new BlobContainerClient(_options.Uri, new DefaultAzureCredential());
```
that would be much more Aspire-friendly.

**Comment by Piedone** [comment]
Hmm, isn't that like half of the solution here, actually (to have it as an option, besides the current key-based auth)? This covers Managed Identity, which is important for all of the managed Azure services. For things lower-level you need Service Principals, but that's a matter of plugging in `ClientSecretCredential` here (and being able to configure it from the config provider).

@ShaneCourtrille?

**Comment by sebastienros** [comment]
I wrote some significant parts of OC and how Aspire registers clients and exposes config, maybe I should look at what you are saying ;)

**Comment by rjpowers10** [comment]
I can spin off a separate issue for Aspire if you want. I tacked on to this issue because if OC was able connect to blob storage with managed identity, I think that would resolve both issues.

I see a related PR from @MikeAlhayek but it looks like it wasn't completed. #18466

**Comment by sebastienros** [comment]
Mike's PR is the best solution right now, but the Azure SDK team is working on a library that will standardize how to configure Azure services in .NET. I had a meeting with the developer a few days ago to ensure it would also work well with Aspire. So maybe something to follow (once it's available publicly).

Found the branch: https://github.com/Azure/azure-sdk-for-net/compare/main...azure-identity-configuration-support
